hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/sink/CleanFunction.java

```diff
@@ -64,7 +64,19 @@ public void open(Configuration parameters) throws Exception {
     this.executor = NonThrownExecutor.builder(LOG).waitForTasksFinish(true).build();
     String instantTime = HoodieActiveTimeline.createNewInstantTime();
     LOG.info(String.format("exec clean with instant time %s...", instantTime));
-    executor.execute(() -> writeClient.clean(instantTime), "wait for cleaning finish");
+    executor.execute(() -> {
+      if (conf.getBoolean(FlinkOptions.CLEAN_ASYNC_ENABLED)) {
+        this.isCleaning = true;
+        try {
+          this.writeClient.clean(instantTime);
+        } catch (Throwable throwable) {
```

Comment:
We need to execute the cleaning no matter whether async cleaning is true or false.

Reply:
Right. Removed the catch clause and `ASYNC_ENABLED` condition, and add `!isCleaning` condition before `clean` function calls in subclass `ClusteringCommitSink` and `CompactionCommitSink`.
Pls take another look, thanks